File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/repository/NoticeRepository.java
package com.windea.demo.cloudcollect.core.repository;

import com.windea.demo.cloudcollect.core.domain.entity.Notice;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface NoticeRepository extends JpaRepository<Notice, Long> {
	Page<Notice> findByUser_Id(Long userId, Pageable pageable);

	long countByUser_Id(Long userId);

	Page<Notice> findByUser_IdAndReadStatus(Long userId, Boolean readStatus, Pageable pageable);

	long countByUser_IdAndReadStatus(Long userId, Boolean readStatus);
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/repository/NoticeRepository.kt
package com.windea.demo.cloudcollect.core.repository

import com.windea.demo.cloudcollect.core.domain.entity.*
import org.springframework.data.domain.*
import org.springframework.data.jpa.repository.*

interface NoticeRepository : JpaRepository<Notice, Long> {
	fun findAllByUserId(userId: Long, pageable: Pageable): Page<Notice>
	
	fun findAllByUserIdAndReadStatus(userId: Long, readStatus: Boolean, pageable: Pageable): Page<Notice>
	
	fun countByUserId(userId: Long): Long
	
	fun countByUserIdAndReadStatus(userId: Long, readStatus: Boolean): Long
}
